packages/components/MenuButton/src/MenuButton.tsx

```diff
@@ -31,6 +31,36 @@ export const MenuButton = compose<MenuButtonType>({
       setShowContextualMenu(!showContextualMenu);
     }, [showContextualMenu, setShowContextualMenu]);
 
+    const menuItemsUpdated = menuItems.map((item) => {
+      if (item.hasSubmenu) {
+        const [showSubmenu, setShowSubmenu] = useState(false);
+        const [isSubmenuVisible, setIsSubmenuVisible] = React.useState(false);
+
+        const toggleShowSubmenu = React.useCallback(() => {
+          setShowSubmenu(!showSubmenu);
+          setIsSubmenuVisible(!isSubmenuVisible);
+        }, [showSubmenu, isSubmenuVisible, setShowSubmenu, setIsSubmenuVisible]);
+
+        const onShowSubmenu = React.useCallback(() => {
+          setIsSubmenuVisible(true);
+        }, [setIsSubmenuVisible]);
+
+        const onDismissSubmenu = React.useCallback(() => {
+          setShowSubmenu(false);
+        }, [setShowSubmenu]);
+        const { onHoverIn = toggleShowSubmenu, submenuProps = {}, ...restItems } = item;
+        const { onShow = onShowSubmenu, onDismiss = onDismissSubmenu, setShowMenu = toggleShowSubmenu, ...restSubmenuProps } = submenuProps;
+        const menuItemUpdated = {
+          ...restItems,
+          onHoverIn,
+          showSubmenu: showSubmenu || item.showSubmenu,
```

Comment:
This code looks like our internal choice of showSubmenu would override the client's input. Usually I would expect the opposite (client can override platform defaults) - is this intended?

Reply:
I checked, client input is used. I added console log to client input and to default methods. For an example with client methods it showed correct info.
![image](https://user-images.githubusercontent.com/11574680/125758240-7c4a4f15-a9c3-42c4-bb92-d48a8bc05b43.png)
![image](https://user-images.githubusercontent.com/11574680/125758300-49d4d745-06c8-47c1-a437-0beb32f85aad.png)
For an example where there's no methods, it uses default ones:
![image](https://user-images.githubusercontent.com/11574680/125758111-4a99bfcd-be13-4c17-b2ca-258e15ab904c.png)
![image](https://user-images.githubusercontent.com/11574680/125758412-b448dbae-5874-4397-9bc4-47e2c0d45ad7.png)

Reply:
Good catch